index.js

```diff
@@ -33,51 +44,107 @@ const Request = {
     const key = Verifier.manifestHash(manifest)
     const id = HypercoreID.normalize(key)
 
+    const isDrive = state.start !== state.end && c.uint8.decode(state) !== 0
+    const blobs = isDrive
+      ? {
+          length: c.uint.decode(state),
+          treeHash: c.fixed32.decode(state)
+        }
+      : null
+
     return {
       version,
       id,
       key,
       length,
       fork,
       treeHash,
-      manifest
+      manifest,
+      isDrive,
+      blobs
     }
   }
 }
 
 module.exports = {
-  async generate (core, { length = core.length, fork = core.fork, manifest = null } = {}) {
-    if (!core.opened) await core.ready()
+  generate,
+  generateDrive,
+  decode,
+  signable
+}
 
-    if (core.core.compat && !manifest) throw new Error('Cannot generate signing requests for compat cores')
-    if (core.fork !== fork) throw new Error('Core should have the same fork')
-    if (!manifest) manifest = core.manifest
+async function generate (core, { length = core.length, fork = core.fork, manifest = null } = {}) {
+  if (!core.opened) await core.ready()
 
-    return c.encode(Request, {
-      version: VERSION,
-      length,
-      fork,
-      treeHash: await core.treeHash(length),
-      manifest
-    })
-  },
-  decode (buffer) {
-    const state = { start: 0, end: buffer.byteLength, buffer }
-    const req = Request.decode(state)
+  if (core.blobs) return generateDrive(core, { length, fork, manifest })
```

Comment:
Just to clarify: this is a check to see if what was passed in was in fact a Hyperdrive object?

Depending on the API we want, we could also detect when the passed-in Hypercore is a Hyperdrive (using https://github.com/holepunchto/hypercore-detector) and generating the Hyperdrive object on the fly. In that case it's no longer straightforward to sign a Hyperdrive's metadata core as just a Hypercore though, so not sure if we want this